### ISSUE TITLE
UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'set' of undefined at SPFetchClient.<anonymous>

### ISSUE THREAD

**Issue by Saintenr** [pnp/pnpjs#1938]
Thank you for reporting an issue, suggesting an enhancement, or asking a question. We appreciate your feedback - to help the team understand your
needs please complete the below template to ensure we have the details to help. Thanks!

**Please check out the [Docs](https://pnp.github.io/pnpjs/) to see if your question is already addressed there. This will help us ensure our documentation covers the most frequent questions.**

### Category
- [ ] Enhancement
- [ x ] Bug
- [ x ] Question
- [ ] Documentation gap/issue

### Version

Please specify what version of the library you are using: [   "@pnp/nodejs-commonjs": "^2.10.0"      ]

Please specify what version(s) of SharePoint you are targeting: [    SP-Online     ]

*If you are not using the latest release, please update and see if the issue is resolved before submitting an issue.*

### Expected / Desired Behavior / Question
I an trying to make a fetch against sp rest api with the SPFetchClient.fetch("url") call.

### Observed Behavior
After setting up the Client with ID and Secret i got this error:
(node:16208) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'set' of undefined
at SPFetchClient.<anonymous> /node_modules/@pnp/nodejs-commonjs/net/spfetchclient.js:42:41)
at step /node_modules/tslib/tslib.js:143:27)
at Object.next /node_modules/tslib/tslib.js:124:57)
at fulfilled /node_modules/tslib/tslib.js:114:62)
at process._tickCallback (internal/process/next_tick.js:68:7)

### Steps to Reproduce
this is made with an azure function on node 12. My Code below:

import { SPFetchClient } from "@pnp/nodejs-commonjs";
const SPClient = new SPFetchClient(
        process.env.SiteUrl, 
        process.env.ClientId,
        process.env.ClientSecret
    );
const r = await SPClient.fetch(apiPath, {});

the apiPath is a string created before the call. The apipath string is working at the browser so this should not the issue.
path example:
https://xxx.sharepoint.com/_api/v2.1/drives/{driveid}/items/{itemid}/getActivitiesByInterval(startDateTime='2020-01',endDateTime='2021-12',interval='month')

thanks for your help

**Comment by juliemturner** [comment]
I would check out this article on working with PnPjs in NodeJS based on the fact that you said your code is running in Azure Function. https://pnp.github.io/pnpjs/nodejs-support/

**Comment by Saintenr** [comment]
i read this article. my tsconfig looks like this: 

{
  "compilerOptions": {
    "module": "commonjs",
    "target": "es6",
    "moduleResolution": "node",
    "declaration": true,
    "outDir": "dist",
    "rootDir": ".",
    "skipLibCheck": true,
    "sourceMap": true,
    "lib": [
      "dom",
      "esnext"
    ],
    "strict": false,
    "esModuleInterop": true
  },
  "files": [
    "./myfunction/index.ts"
  ]
}

and my package.json 

"scripts": {
        "build": "tsc",
        "watch": "tsc -w",
        "prestart": "npm run build",
        "start": "func start",
        "test": "echo \"No tests yet...\""
    },
    "dependencies": {
        "@pnp/nodejs-commonjs": "^2.10.0",
        "@pnp/sp-commonjs": "^2.10.0",
        "jquery": "^3.6.0"
    },
    "devDependencies": {
        "@azure/functions": "^1.2.3",
        "typescript": "^4.0.0"
    }


if i config sp with sp.config and use the sp.* library for my request e.g. list items its working fine. But i want to make a custom get call and tried to use SPFetchClient and the fetch function to do this. 
If this is the wrong way or my config is wrong, maybe you can help me.

thanks a lot

**Comment by Saintenr** [comment]
it seems to be a problem on the spfetchclient.js that headers = undefined.
line 42: --> options.headers.set("Authorization", "Bearer " + token.access_token);

just an idea.

**Comment by Saintenr** [comment]
I fixed the issue on my end. 
The problem was found on the spfetchclient.js file on line 42. The declaration of options is wrong. And the function set() can not be found.

I overhanded the fetch function these option object:

const options: { headers: Headers } = {
                        headers: new Headers,
                    };
await SPClient.fetch(apiPath, options);

now headers is from type Headers and the function set() can be found.
Maybe this can be added to the next version.
